src/sys/pthread.rs

```diff
@@ -34,10 +41,285 @@ pub fn pthread_kill<T>(thread: Pthread, signal: T) -> Result<()>
+            libc::PTHREAD_PRIO_PROTECT => Self::Protect,
+            _ => unreachable!()
+        }
+    }
+}
+
+/// Mutex attributes.
+#[cfg(target_os = "linux")]
+#[derive(Debug)]
+pub struct MutexAttr(libc::pthread_mutexattr_t);
+
+#[cfg(target_os = "linux")]
+impl MutexAttr {
+    /// Wraps [`libc::pthread_mutexattr_init`].
+    pub fn new() -> Result<Self> {
+        let attr = unsafe {
+            let mut uninit = std::mem::MaybeUninit::<libc::pthread_mutexattr_t>::uninit();
+            Errno::result(libc::pthread_mutexattr_init(uninit.as_mut_ptr()))?;
+            uninit.assume_init()
+        };
+        Ok(Self(attr))
+    }
+    /// Wraps [`libc::pthread_mutexattr_getpshared`].
+    pub fn get_shared(&self) -> Result<bool> {
+        let init = unsafe {
+            let mut uninit = std::mem::MaybeUninit::uninit();
+            Errno::result(libc::pthread_mutexattr_getpshared(&self.0,uninit.as_mut_ptr()))?;
+            uninit.assume_init()
+        };
+        Ok(init == libc::PTHREAD_PROCESS_SHARED)
+    }
+    /// Wraps [`libc::pthread_mutexattr_setpshared`].
+    pub fn set_shared(&mut self, shared: bool) -> Result<()> {
+        let shared = if shared { libc::PTHREAD_PROCESS_SHARED} else { libc::PTHREAD_PROCESS_PRIVATE };
+        unsafe {
+            Errno::result(libc::pthread_mutexattr_setpshared(&mut self.0,shared)).map(drop)
+        }
+    }
+    /// Wraps [`libc::pthread_mutexattr_getrobust`].
+    pub fn get_robust(&self) -> Result<bool> {
+        let init = unsafe {
+            let mut uninit = std::mem::MaybeUninit::uninit();
+            Errno::result(libc::pthread_mutexattr_getrobust(&self.0,uninit.as_mut_ptr()))?;
+            uninit.assume_init()
+        };
+        Ok(init == libc::PTHREAD_MUTEX_ROBUST)
+    }
+    /// Wraps [`libc::pthread_mutexattr_setrobust`].
+    pub fn set_robust(&mut self, robust: bool) -> Result<()> {
+        let robust = if robust { libc::PTHREAD_MUTEX_ROBUST} else { libc::PTHREAD_MUTEX_STALLED };
+        unsafe {
+            Errno::result(libc::pthread_mutexattr_setrobust(&mut self.0,robust)).map(drop)
+        }
+    }
+    /// Wraps [`libc::pthread_mutexattr_getprotocol`].
+    pub fn get_protocol(&self) -> Result<Protocol> {
+        let init = unsafe {
+            let mut uninit = std::mem::MaybeUninit::uninit();
+            Errno::result(libc::pthread_mutexattr_getprotocol(&self.0,uninit.as_mut_ptr()))?;
+            uninit.assume_init()
+        };
+        Ok(Protocol::from(init))
+    }
+    /// Wraps [`libc::pthread_mutexattr_setprotocol`].
+    pub fn set_protocol(&mut self, protocol: Protocol) -> Result<()> {
+        unsafe {
+            Errno::result(libc::pthread_mutexattr_setprotocol(&mut self.0,protocol as i32)).map(drop)
+        }
+    }
+}
+
+#[cfg(target_os = "linux")]
+impl Default for MutexAttr {
+    fn default() -> Self {
+        let mutex_attr = Self::new().unwrap();
+        debug_assert_eq!(mutex_attr.get_shared(),Ok(true));
```

Comment:
Why is this line here?  From my reading of https://linux.die.net/man/3/pthread_mutexattr_init , the opposite is true.  Mutexes are private by default.  I wouldn't trust that to be the same across all OSes, though.  And why use debug_assert_eq instead of assert_eq ?

Reply:
Removed.